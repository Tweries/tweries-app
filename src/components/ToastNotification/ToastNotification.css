.ToastNotification {
  border: 1px solid;
  border-radius: 4px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 32px;
  width: 100%;
}

.ToastNotification button {
  background: transparent;
  border: 0;
  margin: 8px 8px 8px 0;
  width: 32px;
}

.ToastNotification span {
  margin: 8px 0 8px 8px;
}

.ToastNotification--danger {
  background-color: #f8d7da;
  border-color: #f5c6cb;
  color: #721c24;
}

.ToastNotification--success {
  background-color: #d4edda;
  border-color: #c3e6cb;
  color: #155724;
}
