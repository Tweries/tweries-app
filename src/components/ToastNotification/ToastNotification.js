import classnames from 'classnames';
import React from 'react';

function ToastNotification({ notification, onClick }) {
  if (notification) {
    const { message, type } = notification;
    return (
      <div
        className={classnames('border flex justify-between mb-2 p-2 rounded', {
          'bg-green-100 border-green-800 text-green-800': type === 'success',
          'bg-red-100 border-red-800 text-red-800': type === 'danger'
        })}
      >
        <span>{message}</span>
        <button className="font-bold" data-testid="dismiss" onClick={onClick}>
          &times;
        </button>
      </div>
    );
  } else {
    return null;
  }
}

export default ToastNotification;
