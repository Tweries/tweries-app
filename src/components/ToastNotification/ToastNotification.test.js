import { render } from '@testing-library/react';
import React from 'react';
import ToastNotification from './ToastNotification';

describe('ToastNotification', () => {
  it('empty', () => {
    const { container } = render(<ToastNotification onClick={jest.fn()} />);

    expect(container).toMatchSnapshot();
  });

  it('success', () => {
    const { container } = render(
      <ToastNotification
        notification={{
          link:
            'https://twitter.com/1183836409850814464/status/1230233654959407104',
          type: 'success'
        }}
        onClick={jest.fn()}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('danger', () => {
    const { container } = render(
      <ToastNotification
        notification={{ message: 'Oh Noes!', type: 'danger' }}
        onClick={jest.fn()}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
