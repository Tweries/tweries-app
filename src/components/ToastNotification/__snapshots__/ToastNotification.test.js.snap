// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToastNotification /> danger 1`] = `
<div>
  <p
    class="border flex justify-between mb-2 p-2 rounded bg-red-100 border-red-800 text-red-800"
  >
    <span>
      Oh Noes!
    </span>
    <button
      class="font-bold"
      data-testid="dismiss"
      type="button"
    >
      ×
    </button>
  </p>
</div>
`;

exports[`<ToastNotification /> empty 1`] = `<div />`;

exports[`<ToastNotification /> success 1`] = `
<div>
  <p
    class="border flex justify-between mb-2 p-2 rounded bg-green-100 border-green-800 text-green-800"
  >
    <span>
      <a
        class="underline"
        href="https://twitter.com/1183836409850814464/status/1230233654959407104"
        rel="noopener noreferrer"
        target="_blank"
      >
        Your tweetstorm has been created!
      </a>
    </span>
    <button
      class="font-bold"
      data-testid="dismiss"
      type="button"
    >
      ×
    </button>
  </p>
</div>
`;
